s3-lambda-bedrock-aoss/src/document_ingestion.py

```diff
@@ -0,0 +1,95 @@
+import json
+import boto3 
+import os
+from opensearchpy import OpenSearch, RequestsHttpConnection, AWSV4SignerAuth
+from PyPDF2 import PdfReader
+
+
+#copy an object from s3 to /tmp
+def copy_object_to_tmp(bucket, key):
+    s3_client = boto3.client("s3")
+    file_name = os.path.basename(key)
+    local_path = f"/tmp/{file_name}"
+    s3_client.download_file(bucket, key, local_path)
+    print(f"Downloaded file to local path {local_path}")
+    return local_path
+
+#create a list variable named pages, use PdfReader to extract pages, enumerating over the pages collecting the index and page. Then add an item to the pages list, each item to include a page number and the page text
+def extract_pages_from_pdf(local_path):
+    reader = PdfReader(local_path, 'rb')
+    pages = []
+    for index, page in enumerate(reader.pages):
+        pages.append([index, page.extract_text()])
+    print(f"Extracted {len(pages)} pages")
+    return pages
+
+#with input as a page list containing page numbers and page text, write a function to embed the text and return the embedding, save the embedding, page number and page text to Amazon OpenSearch
+def embed_and_store_pages(pages):
+    responses=[]
+    for page in pages:
+        embedding = create_embeddings(content=page[1])
+        print(f"Created embeddings for page #{page[0]+1}")
+        response = index_document_page(page_number=page[0], page_text=page[1], embedding=embedding)
+        print(f"Indexed content for page #{page[0]+1}")
+        responses.append(response)
+    return responses
+
+
+def create_embeddings(content, modelId="amazon.titan-embed-text-v1"):
+    bedrock_client = boto3.client("bedrock-runtime", region_name=os.environ["BEDROCK_MODEL_REGION"])
+    accept = "application/json"
+    contentType = "application/json"
+    body = json.dumps({"inputText": content})
+    response = bedrock_client.invoke_model(
+        body=body, modelId=modelId, accept=accept, contentType=contentType
+    )
+    response_body = json.loads(response.get("body").read())
+    embedding = response_body.get("embedding")
+    return embedding
+
+
+#write a function to remove a prefix of https:// from the given string
```

Comment:
The comments doesn't match the code that's already written. e.g. could the comment here be something like "function to remove a prefix 'https://' from the given string" ?
Please check the same with other comments/files in the PR. Thanks.

Reply:
fixed